### PR TITLE
Add an example to add custom metrics on proxy wasm

### PR DESCRIPTION
**Pull request by alandiegosantos**

### CODE REVIEW

File: examples/http_metrics.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+pub trait CounterTrait {
+    /// Increments counter by `1`.
+    fn inc(&self) -> Result<(), Status> {
+        self.add(1)
+    }
+    /// Increments counter by a given offset.
+    fn add(&self, offset: u64) -> Result<(), Status>;
+    /// Returns current value of the counter.
+    fn value(&self) -> Result<u64, Status>;
+}
```
Comment (PiotrSikora) [inline examples/http_metrics.rs:94]: Should this trait (with default implementation from above) be added to `src/traits.rs` instead?

**Review by PiotrSikora — COMMENTED** [review]
You need to add those examples to `Cargo.toml` to pass the CI.

File: examples/http_metrics.rs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+fn counter(name: &str) -> Result<Box<dyn CounterTrait>, Status> {
+    hostcalls::define_metric(MetricType::Counter, name)
+        .map(|metric_id| Box::new(Counter(metric_id)) as Box<dyn CounterTrait>)
+}
```
[the remark this answers is not in the document]
Comment (alandiegosantos) [reply examples/http_metrics.rs:99]: Envoy is smart in this topic and increment the right metric. I agree with moving this to the startup. I will take a look at it later.

File: examples/http_metrics.rs (hunk 1 above)
Comment (alandiegosantos) [reply examples/http_metrics.rs:94]: `http_metrics.rs `was supposed to be just an example of how to use metrics API. 
CounterTrait can be moved to `src/traits.rs` along with Gauge and Histogram implementations, adding some abstraction to the code.
However, should this lib be as close as possible from proxy-wasm spec?

**Pull request closed without merging**